cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -122,16 +116,16 @@
   /**
    * Json serializer.
    */
-  private static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(new GsonBuilder())
+  protected static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(new GsonBuilder())
     .registerTypeAdapter(ScheduleSpecification.class, new ScheduleSpecificationCodec())
     .create();
 
   /**
    * Store manages non-runtime lifecycle.
    */
-  private final Store store;
+  protected final Store store;
 
-  private final WorkflowClient workflowClient;
+  protected final WorkflowClient workflowClient;
```

Comment:
Why protected? If this field is only use by the sub-class (WorkflowHttpHandler), then do the guice injection there.